File: ruby/raindrops/raindrops.rb
module BookKeeping
  VERSION = 3
end

class Raindrops
  TABLE = { 3 => "Pling",
            5 => "Plang",
            7 => "Plong" }

  def self.convert(num)
    result = TABLE.select do |key, word|
      (num % key).zero?
    end.values
    result.empty? ? num.to_s : result.join() # implicit return
  end
end

File: cpp/raindrops/raindrops.cpp
#include "raindrops.h"
#include <sstream>
#include <vector>

namespace raindrops {
const std::vector<std::pair<unsigned int, std::string>> mapping = {
    {3, "Pling"}, {5, "Plang"}, {7, "Plong"}};
}

std::string raindrops::convert(const unsigned int number) {
  std::stringstream result;
  for (const auto& kv : raindrops::mapping)
    if (number % kv.first == 0)
      result << kv.second;
  return (result.str().empty()) ? std::to_string(number) : result.str();
}
